analyze.go

```diff
@@ -46,6 +51,42 @@ func getKindFor(v cue.Value) (TSType, error) {
 	return TSType(tt), nil
 }
 
+func getKindForField(v cue.Value) (TSType, bool) {
+	if field, ok := v.Source().(*ast.Field); ok {
```

Comment:
The overall approach seems like at least an iterative improvement and therefore OK. But i get nervous every we rely on `cue.Value.Source()`, because i don't fully understand the cases when we will or won't get a `nil` return, and therefore i worry that we could end up with inconsistent user-facing behavior.

Reply:
We still in experimental stage. We can remove in future releases if it doesn't worth it.